test/metabase/db/custom_migrations_test.clj

```diff
@@ -2546,3 +2547,104 @@
+                                                                  :enabled      true})))]
+
+            (testing "before migration there should be a trigger for the send pulse task"
+              (task/init! ::task.send-pulses/SendPulses)
+              (is (=? [{:key (format "metabase.task.send-pulse.trigger.%d.0_0_18_*_*_?_*" pulse-id)}]
+                      (u/poll {:thunk       send-pulse-triggers
+                               :done?       seq
+                               :interval-ms 20}))))
+
+            (testing "after migration"
+              (migrate!)
+              (testing "pulse is migrated to notification"
+                (let [notification (t2/select-one :notification)
+                      notification-card (t2/select-one :notification_card :id (:payload_id notification))
+                      subscription (t2/select-one :notification_subscription :notification_id (:id notification))
+                      handler      (t2/select-one :notification_handler :notification_id (:id notification))
+                      recipient    (t2/select-one :notification_recipient :notification_handler_id (:id handler))]
+                  (is (= {:payload_type "notification/card"
+                          :active       true
+                          :creator_id   user-id}
+                         (select-keys notification [:payload_type :active :creator_id])))
+
+                  (is (= {:card_id        card-id
+                          :send_once      false
+                          :send_condition "has_result"}
+                         (select-keys notification-card [:card_id :send_once :send_condition])))
+
+                  (is (= {:type          "notification-subscription/cron"
+                          :cron_schedule "0 0 18 * * ? *"}
+                         (select-keys subscription [:type :cron_schedule])))
+
+                  (is (= {:channel_type "channel/email"}
+                         (select-keys handler [:channel_type])))
+                  (is (= {:type    "notification-recipient/raw-value"
+                          :details "{\"value\":\"test@test.com\"}"}
+                         (select-keys recipient [:type :details])))
+                  (is (= {:type    "notification-recipient/raw-value"
+                          :details "{\"value\":\"test@test.com\"}"}
+                         (select-keys recipient [:type :details])))
+
+                  (testing "send pulse trigger is deleted"
+                    (is (empty? (send-pulse-triggers))))
+
+                  (testing "init send notification creates new triggers"
+                    ;; simulate start up
+                    (task/init! ::task.notification/SendNotifications)
+                    (is (=? [{:key (format "metabase.task.notification.trigger.subscription.%d" (:id subscription))}]
+                            (u/poll {:thunk       send-notification-triggers
```

Comment:
Worth asserting that there were no triggers before you initialized the `SendNotifications` task?